pkg/util/pki/nameconstraints.go

```diff
@@ -32,86 +37,532 @@ var (
 
 // NameConstraints represents the NameConstraints extension.
 type NameConstraints struct {
-	PermittedDNSDomainsCritical bool        `asn1:"optional,explicit,tag:0"`
-	PermittedDNSDomains         []string    `asn1:"optional,explicit,tag:1"`
-	ExcludedDNSDomains          []string    `asn1:"optional,explicit,tag:2"`
-	PermittedIPRanges           []net.IPNet `asn1:"optional,explicit,tag:3"`
-	ExcludedIPRanges            []net.IPNet `asn1:"optional,explicit,tag:4"`
-	PermittedEmailAddresses     []string    `asn1:"optional,explicit,tag:5"`
-	ExcludedEmailAddresses      []string    `asn1:"optional,explicit,tag:6"`
-	PermittedURIDomains         []string    `asn1:"optional,explicit,tag:7"`
-	ExcludedURIDomains          []string    `asn1:"optional,explicit,tag:8"`
+	PermittedDNSDomainsCritical bool
+	PermittedDNSDomains         []string
+	ExcludedDNSDomains          []string
+	PermittedIPRanges           []*net.IPNet
+	ExcludedIPRanges            []*net.IPNet
+	PermittedEmailAddresses     []string
+	ExcludedEmailAddresses      []string
+	PermittedURIDomains         []string
+	ExcludedURIDomains          []string
 }
 
 // Adapted from x509.go
 func MarshalNameConstraints(nameConstraints *v1.NameConstraints) (pkix.Extension, error) {
```

Comment:
Can we use `NameConstraints` as input instead of `v1.NameConstraints`?
This will decouple this Marshal function from the kubernetes API.

Reply:
Done